doc/pages/development-guide/001-dev_guide.md

```diff
@@ -0,0 +1,165 @@
+
+- You have read the [contributing guidelines](https://github.com/astarte-platform/astarte/blob/master/CONTRIBUTING.md) for the project
+- You are familiar with basic Astarte concepts such as Realms, Interfaces, Devices and Triggers. If you are not, you can learn more on the [documentation](001-intro_architecture.md)!
+- You have already run at least once [Astarte in 5 minutes](010-astarte_in_5_minutes.md)
+- If you plan to work on the code, you have some knowledge of Elixir and Docker
+
+## Where to start?
+
+To make your contribution, let's first identify where to do it.
+
+### Code
+
+In general, you will probably touch one of the Astarte components.
+Astarte is made up of a number of different microservices: [here](020-components.html) you can find a quick summary of each.
+
+Moreover, components that expose an API (Housekeeping, Pairing, Realm Management) are split in an API service and a backend one. All of those are located in the [main Astarte repository](https://www.github.com/astarte-platform/astarte) on Github, in the `apps` directory.
+
+Finally, Astarte is backed by three libraries:
+
+- [Astarte Core](https://github.com/astarte-platform/astarte_core): contains definitions common to the application, e.g. Realm name, Device ID validity, Interface definitions etc.
+- [Astarte RPC](https://github.com/astarte-platform/astarte_rpc): contains all the Protobufs and code used in the internal RPC mechanism. In most cases, RPCs are between an API component and the related backend.
+- [Astarte Data Access](https://github.com/astarte-platform/astarte_data_access): contains some interactions with the database that are common to all Astarte services, e.g. queries for interface values.
+
+### Documentation
+
+Documentation is automatically generated from files in the `doc/pages` subdirectory of the [main Astarte repository](https://www.github.com/astarte-platform/astarte) on Github.
+It is written in Github-flavored markdown.
+
+### Tooling
+
+The main Astarte repo contains a number of tools that can be used everyday when interacting with Astarte. They are found in the `tools` directory subdirectory of the [main Astarte repository](https://www.github.com/astarte-platform/astarte) on Github:
+
+- Astarte Device Fleet Simulator: create a fleet of virtual devices. Mainly used for load testing
+- Astarte E2E: a tool that sends data from a virtual device and checks the resulting value in Astarte using triggers. Mainly used for monitoring
+- Astarte Import and Astarte Export: used to load/export data into/from Astarte
+
+## Can I use an Elixir REPL to help me during development?
+
+Sure thing! However, usually Astarte services do need some context around them.
+Refer to the following [”Testing a single component”](#testing-a-single-component) paragraph for dependencies, then you can start your application with an interactive shell by replacing `mix test` with `iex -S mix`
```

Comment:
Do we have a standard procedure for testing a single component yet? or is it a WIP?

Reply:
Yup, it's some lines below

Reply:
alright, so anchor ref is broken

Reply:
No, once it passes through `mix docs` everything will be all right, see e.g. the last line [here](https://docs.astarte-platform.org/astarte/latest/030-interface.html#datastream) which is just
https://github.com/astarte-platform/astarte/blob/fb5657d06ebfc806d944cce354e261c18796960d/doc/pages/architecture/030-interface.md?plain=1#L93